service/resource/cloudformation/template.go

```diff
@@ -1,18 +1,22 @@
 package cloudformation
 
 const (
-	MainTemplate = `AWSTemplateFormatVersion: 2010-09-09
-Description: {{ .ASGType }} autoscaling group
+	MainTemplate = `{{define "main"}}AWSTemplateFormatVersion: 2010-09-09
+Description: Main CloudFormation stack.
 Resources:
-  {{ .ASGType }}LaunchConfiguration:
+  {{template "launch_configuration" .}}
+  {{template "autoscaling_group" .}}
+{{end}}`
+
+	LaunchConfigurationTemplate = `{{define "launch_configuration"}}{{ .ASGType }}LaunchConfiguration:
     Type: "AWS::AutoScaling::LaunchConfiguration"
     Description: {{ .ASGType }} launch configuration
     Properties:
-      ImageId: !Ref ImageID
+      ImageId: {{ .ImageID }}
```

Comment:
This templating in the resources is nice and much less cumbersome than all the parameters <3